plugin/src/main/java/org/xtclang/plugin/tasks/XtcTestTask.java

```diff
@@ -0,0 +1,112 @@
+package org.xtclang.plugin.tasks;
+
+import static org.xtclang.plugin.XtcPluginConstants.XTC_LAUNCHER_CLASS_NAME;
+
+import javax.inject.Inject;
+
+import org.gradle.api.Project;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.process.ExecResult;
+
+import org.xtclang.plugin.XtcRunModule;
+import org.xtclang.plugin.XtcRuntimeExtension;
+import org.xtclang.plugin.launchers.CommandLine;
+import org.xtclang.plugin.launchers.XtcLauncher;
+
+/**
+ * Task that runs and XTC module, given at least its name, using the module path from
+ * the XTC environment.
+ * <p>
+ * We add a default run task to an XTC project. The default is either run nothing, or run everything.
+ * First it looks in xtcRun and tries to run the modules from there.
+ *    If no modules are there, we just say "nothing to run"
+ *    Later - support command line modules, if we don't want to just keep that logic in build scripts, which is fine and can be lazy too.
+ *    If there is a module command line, we replace the xtcRun modules with these, and we leave the tasks alone.
+ *    They still override, but if you haven't touched the default xtcRun task that does what you want, i.e. runXtc -PmoduleName=foo
+ * If there is a module config in the task, it overrides/replaces those totally.
+ * <p>
+ * We should easily be able to create XTC run tasks of our own
+ * If the 'runXtc' task is called without any module specifications, it will try to run every module compiled
+ * from the source set in undefined order. That would be the exact equivalent of calling 'runAllXtc'
+ * (which does not need to be configured, and is a task added by the plugin).
+ */
+
+// TODO: Add modules {} segment to runtime DSL
+// TODO: Add a generic xtcPlugin or xtc extension, where we can set stuff like, e.g. log level for the plugin (which does not redirect)
+// TODO: Add WorkerExecutor and the Gradle Worker API to execute in parallel if there are no dependencies.
+//   Any task with zero defined outputs is not cacheable, which should be enough for all run tasks.
+// TODO: Make the module path/set pattern filterable for the module DSL.
+public abstract class XtcTestTask extends XtcBaseRunTask implements XtcRuntimeExtension {
```

Comment:
Looks good. Let me know if you need help figuring out any DSL modifications, it should be relatively staightforward by looking at what is there already, but there are some hidden traps. 

One thing that we want to do is to run the Gradle configuration cache, which will significantly speed up the pre-build phase of the gradle jobs - this requires that we do not store projects anywhere in the plugin, which is possible to do, but requires some rewrites.  Enabling the configuration cache will tell us what goes wrong and why, and I was hoping to get a volunteer to this fairly self contained task.